src/rosdiscover/cli.py

```diff
@@ -83,6 +86,90 @@ def rosservice_list(args) -> None:
     print('\n'.join(sorted(services)))
 
 
+def toString(line):
+    s = ''
+    l1 = list(line)
+    if len(l1) == 0:
+        return '[]'
+    for i in l1:
+        (fmt, topic) = i
+        s = s + "\n  - format: " + fmt + "\n    name: " + topic + "\n"
+    return s
+
+
+def get_info(image, sources, environment, file, package):
+    rsw = roswire.ROSWire()
+    with rsw.launch(image, sources, environment=environment) as system:
+        with system.roscore() as ros:
+            ros.roslaunch(file,
+                          package=package,
+                          args={'gui': 'false'})
+
+            time.sleep(30)
+            node_names = list(ros.nodes)
+            state = ros.state
+            topic_to_type = ros.topic_to_type
+            service_to_format = {}
+            for service_name in state.services:
+                service = ros.services[service_name]
+                service_to_format[service_name] = service.format.name
+
+    return node_names, state, topic_to_type, service_to_format
+
+
+def create_dict(node_names, state, topic_to_type, service_to_format):
+    nodeSummaryDict = {}
+    for n in node_names:
+        p = []
+        for key in state.publishers:
+            pubs = state.publishers[key]
+            for i in pubs:
+                if i == n:
+                    p.append((topic_to_type[key], key))
+        s = []
+        for key in state.subscribers:
+            subs = state.subscribers[key]
+            for i in subs:
+                if i == n:
+                    s.append((topic_to_type[key], key))
+        serv = []
+        for key in service_to_format:
+            path = n + '/'
+            if path in key:
+                serv.append((service_to_format[key], key))
+        obj = NodeSummary('', n, '/', '', '', False, '', False, p, s, [], [], [], serv, [], [])
+        nodeSummaryDict.update({n: obj})
+    return nodeSummaryDict
+
+
+def dynamic_analysis(args):
+    print(args)
+    with open(args.config.name, 'r') as r:
+        data = yaml.safe_load(r)
+    r.close()
+    f = open("Arch.yml", "w")
```

Comment:
This should be passed in as a parameter, which defaults to `arch,yml`. Add it as `--output` to the arg parser.

Reply:
Also, no need to explicitly `r.close()` here. `with` automatically takes care of calling `open` and `close` for you :slightly_smiling_face: (`r` will be closed upon reaching the end of the `with` block`.)

Reply:
And, in general, use `with open(...) as f:` every time that you perform file operations.